zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/user/UpdateUserTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.user;
+
+import static io.camunda.zeebe.protocol.record.Assertions.assertThat;
+
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.util.UUID;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.jupiter.api.DisplayName;
+
+public class UpdateUserTest {
+  @ClassRule public static final EngineRule ENGINE = EngineRule.singlePartition();
+
+  @Rule
+  public final RecordingExporterTestWatcher recordingExporterTestWatcher =
+      new RecordingExporterTestWatcher();
+
+  @DisplayName("should update a user if the username exists")
+  @Test
+  public void shouldUpdateAUser() {
+    final var username = UUID.randomUUID().toString();
+    ENGINE
+        .user()
+        .newUser(username)
+        .withName("Foo Bar")
+        .withEmail("foo@bar.com")
+        .withPassword("password")
+        .create();
+
+    final var updatedUser =
+        ENGINE
+            .user()
+            .updateUser(username)
+            .withName("Bar Foo")
+            .withEmail("foo@bar.blah")
+            .withPassword("Foo Bar")
+            .update();
+
+    final var createdUser = updatedUser.getValue();
+    assertThat(createdUser)
```

Comment:
```suggestion
    final var updatedUser =
        ENGINE
            .user()
            .updateUser(username)
            .withName("Bar Foo")
            .withEmail("foo@bar.blah")
            .withPassword("Foo Bar")
            .update()
            .getValue();

    assertThat(updatedUser)
```

🔧 The naming is a bit strange as it reads as if we're asserting over the createdUser instead of the updatedUser.

Reply:
Its definitely very strange, it should be the updated user :D